### PR TITLE
Reland "Expose API for resizing image caches #41415"

### LINKED ISSUE #26194: Expose an API to resize an Image
This doesn't need to be done automatically, but adding the capability to resize a known large image into a fixed size could help memory usages and performance. In https://github.com/flutter/flutter/issues/21571, using the full size image returned from the image picker can result in OOM.

For example, in Android I can use [ThumbnailUtils](https://developer.android.com/reference/android/media/ThumbnailUtils) to create video or image thumbnails at a particular resolution. (Though this still requires loading the full image into memory).

Better yet, could we resize an image known to be too large to load fully into memory into a small thumbnail with going OOO?

cc @chinmaygarde @Hixie

### PR DESCRIPTION
**Pull request by GaryQian**
Disables the tests that does not apply for web to fix web test breakage. Web does not support this feature, so this test should not run on web.

Reverts https://github.com/flutter/flutter/pull/42764

Original PR: https://github.com/flutter/flutter/pull/41415

This PR creates a new image provider ResizedImage that allows the user to specify the dimensions of the image that the engine can then use to resize to. This is the engine side PR accounting for the same changes: flutter/engine#8596

Fixes #26194

### CODE REVIEW

File: packages/flutter/lib/src/widgets/image.dart [hunk 1]
```diff
@@ -305,6 +323,16 @@ class Image extends StatefulWidget {
...
   ///
   /// If [excludeFromSemantics] is true, then [semanticLabel] will be ignored.
+  ///
+  /// If [cacheWidth] or [cacheHeight] are provided, it indicates to the
+  /// engine that the image should be decoded at the specified size. The image
+  /// will be rendered to the constraints of the layout or [width] and [height]
+  /// regardless of these parameters. These parameters are primarily intended
+  /// to reduce the memory usage of [ImageCache].
+  ///
+  /// In the case where the network image is on the Web platform, the [cacheWidth]
```
Comment (GaryQian) [inline packages/flutter/lib/src/widgets/image.dart]: See here for description on web behavior

Comment (yjbanov) [reply packages/flutter/lib/src/widgets/image.dart]: LGTM, but let's file an issue for us to implement it later. We can definitely implement it in the CanvasKit-based backend, and with some tricks in the HTML backend as well.

**Review by yjbanov — APPROVED** [review]

**Comment by GaryQian** [comment]
Added TODOs and filed https://github.com/flutter/flutter/issues/42789 to track it.

**Comment by sooxiaotong** [comment]
@GaryQian How about FadeInImage with `cacheWidth` and `cacheHeight`?

**Comment by DyaryRaoof** [comment]
I am a bit confused a bit guys. what should i do to fix this ? 
should i update flutter ?

**Comment by GaryQian** [comment]
@DyaryRaoof Are you broken by this change? Or are you asking how to use the new feature?

**Comment by DyaryRaoof** [comment]
@GaryQian No I am not broken by the change , i was redirected to this page as a solution to this issue : https://github.com/flutter/flutter/issues/32143#issuecomment-544303867

And I am not sure what to do next :update flutter ? or is do something else to have the issue solved.

**Comment by GaryQian** [comment]
This is feature is currently on the master and dev channels. You need to update Flutter to the latest version to use this feature.

First, change to the dev channel:
`flutter channel dev`

Then, `flutter upgrade`

You should now be able to specify a `cacheWidth` and `cacheHeight` in your `Image` constructor.

**Comment by sooxiaotong** [comment]
@GaryQian How about FadeInImage with cacheWidth and cacheHeight?I don't know how to write my own, the image provider seems hard for me to understand

**Comment by GaryQian** [comment]
@sooxiaotong Thanks for pointing that out, I'll see if I can add it!

**Comment by GaryQian** [comment]
Correction, this change has not rolled into dev channel yet, please use `flutter channel master` as of now.

**Comment by DyaryRaoof** [comment]
@GaryQian any idea when it will become available ?

**Comment by GaryQian** [comment]
https://github.com/flutter/flutter/pull/43286 is the PR that adds this feature.

**Comment by f-martinez11-loro** [comment]
@DyaryRaoof Did you manage to fix this issue #32143? I'm having the same issue, but I'm not sure if this is the correct fix. I think is more a workaround.

**Comment by DyaryRaoof** [comment]
> @DyaryRaoof Did you manage to fix this issue #32143? I'm having the same issue, but I'm not sure if this is the correct fix. I think is more a workaround.

Not really. the CahedWidth and CachedHeight were supposed to fix the issue. but I still cant find it in the latest version of flutter

**Comment by GaryQian** [comment]
This has been in the master channel for a few days now. Are you sure you are on master channel? (`flutter channel master`). New features and fixes first end up on master channel, then dev, beta, and finally stable. This gives it time for validation by increasingly larger groups of developers.

**Comment by nightfallsad** [comment]
> This has been in the master channel for quite a few days now. Are you sure you are on master channel? (`flutter channel master`). New features and fixes first end up on master channel, then dev, beta, and finally stable. This gives it time for validation by increasingly larger groups of developers. I believe as of now it has not made it into dev channel and above yet.

Will this PR be merge to the current last stable version v1.9.1-hotfixes？

**Comment by GaryQian** [comment]
@nightfallsad It is unlikely that this will be hotfixed in as it is not a critical or crash-fixing change. You are welcome to use this feature on our master and dev channels though!

**Comment by nightfallsad** [comment]
> @nightfallsad It is unlikely that this will be hotfixed in as it is not a critical or crash-fixing change. You are welcome to use this feature on our master and dev channels though!

Ok,thx.

**Comment by DyaryRaoof** [comment]
It's no longer available in the master channel. Not sure about the dev channel though.

**Comment by GaryQian** [comment]
@DyaryRaoof I just double check, it is definitely in master. Are you sure you are on the master channel and have upgraded to the latest version?

Make sure you run `flutter upgrade` to download the latest version of Flutter.

You can see the feature documented in the master channel api docs: https://master-api.flutter.dev/flutter/widgets/Image-class.html

**Comment by DyaryRaoof** [comment]
@GaryQian It's present on the master channel. I deleted flutter SDK , reinstalled , then switched to master to see the change.

**Pull request merged**